app/routes/events/components/StripeElement.js

```diff
@@ -4,38 +4,153 @@ import {
   injectStripe,
   PaymentRequestButtonElement,
   Elements,
-  StripeProvider
+  StripeProvider,
+  CardCVCElement,
+  CardExpiryElement,
+  CardNumberElement
 } from 'react-stripe-elements';
 import config from 'app/config';
-import StripeCheckout from 'react-stripe-checkout';
-import Button from 'app/components/Button';
-import styles from './StripeElement.css';
-import logoImage from 'app/assets/kule.png';
+import stripeStyles from './Stripe.css';
 import type { EventRegistrationChargeStatus, User, Event } from 'app/models';
+import LoadingIndicator from 'app/components/LoadingIndicator';
 
-import { paymentPending } from '../utils';
+type Stripe = {
+  paymentRequest: Object => Object,
+  handleCardPayment: string => Promise<*>,
+  confirmPaymentIntent: (string, Object) => Promise<*>
+};
 
 type Props = {
   event: Event,
   currentUser: User,
-  onToken: (token: string) => Promise<*>,
+  createPaymentIntent: () => Promise<*>,
   chargeStatus: EventRegistrationChargeStatus
 };
 
-type FormProps = Props & { stripe: { paymentRequest: Object => Object } };
+/*
+ * Taken from https://stripe.com/docs/api/errors
+ */
+type StripeError = {
+  type: string,
+  charge: string,
+  code: string,
+  decline_code: string,
+  message: string,
+  param: string,
+  doc_url: string,
+  payment_intent: Object,
+  setup_intent: Object,
+  source: Object
+};
+
+type FormProps = Props & {
+  fontSize?: number
+};
+
+type CardFormProps = FormProps & {
+  ledgend: string,
+  setError: StripeError => void,
+  setSuccess: () => void,
+  setLoading: boolean => void,
+  stripe: Stripe
+};
+
+type PaymentRequestFormProps = FormProps & {
+  setError: StripeError => void,
+  setSuccess: () => void,
+  setLoading: boolean => void,
+  setPaymentRequest: boolean => void,
+  stripe: Stripe
+};
+
+type FormState = {
+  error?: StripeError | null,
+  success?: boolean,
+  loading: boolean,
+  paymentRequest: boolean
+};
 
 type State = {
-  paymentRequest: Object,
+  paymentRequest?: Object,
   canMakePayment?: boolean
 };
 
-const paymentButtonWidth = 130;
+const createOptions = {
+  style: {
+    base: {
+      color: '#424770',
+      letterSpacing: '0.025em',
+      fontFamily: 'Source Code Pro, monospace',
+      '::placeholder': {
+        color: '#aab7c4'
+      }
+    },
+    invalid: {
+      color: '#9e2146'
+    }
+  }
+};
 
-class PaymentRequestForm extends React.Component<FormProps, State> {
+class CardForm extends React.Component<CardFormProps, FormState> {
+  handleSubmit = async ev => {
+    ev.preventDefault();
+    const { stripe, createPaymentIntent } = this.props;
+    if (stripe) {
+      const { payload } = await createPaymentIntent();
```

Comment:
What if creating payment intent fails?

Reply:
You will get a toast with "Betaling feilet", and you can retry the payment. So there is room for improvement with some better error messages for sure.